src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -900,32 +900,79 @@ private void visitProperties(ObjectType objType, boolean isStatic, Set<String> s
-          emitBreak();
-          continue;
-        }
-        if (isStatic) {
-          // The static methods apply and call are provided by lib.d.ts.
-          if (propName.equals("apply") || propName.equals("call")) continue;
-          emit("static");
-        }
-        emit(propName);
-        if (propertyType.isFunctionType()) {
-          visitFunctionDeclaration((FunctionType) propertyType);
-        } else {
-          visitTypeDeclaration(propertyType, false);
-        }
-        emit(";");
+        visitProperty(propName, objType, isStatic);
+      }
+    }
+
+    private void visitProperty(String propName, ObjectType objType, boolean isStatic) {
+      JSType propertyType = objType.getPropertyType(propName);
+      // Some symbols might be emitted as provides, so don't duplicate them
+      String qualifiedName = objType.getDisplayName() + "." + propName;
+      if (provides.contains(qualifiedName)) {
+        return;
+      } else if (isPrivate(objType.getOwnPropertyJSDocInfo(propName))) {
+        return;
+      } else if (propertyType.isEnumType()) {
+        // For now, we don't emit static enum properties. We theorize it should not be needed.
+        emit("/* not emitting " + propName + " because it is an enum and it is not provided */");
+        emitBreak();
+        return;
+      }
+      if (isStatic) {
+        // The static methods apply and call are provided by lib.d.ts.
+        if (propName.equals("apply") || propName.equals("call")) return;
+        emit("static");
+      }
+      emit(propName);
+      if (propertyType.isFunctionType()) {
+        visitFunctionDeclaration((FunctionType) propertyType);
+      } else {
+        visitTypeDeclaration(propertyType, false);
+      }
+      emit(";");
+      emitBreak();
+      if (isStatic) {
+        emitStaticOverloads(propName, objType, propertyType);
+      }
+    }
+
+    /**
+     * Emit overloads for static methods from super classes that do not match the current function.
+     *
+     * In Closure, static methods are completely independent of each other, like regular functions
+     * with different names. In TypeScript, the static side of a class has to extend its parent's
+     * static side, i.e. static methods are inherited and have to subtype the parent's methods.
+     * 
+     * E.g. the following code fails as Child incorrectly extends Parent.
+     * <pre>
+     *   class Parent               { static foo(a: string); }
+     *   class Child extends Parent { static foo(a: number); }
+     * </pre>
+     *
+     * To fix, the code below re-emits (by recursion) any properties that have a colliding type.
+     * 
+     * This is necessarily incomplete, and will fail for differences that cannot be dispatched on,
+     * e.g. different return types. It also pretends that functions exist that do not.
+     */
+    private void emitStaticOverloads(String propName, ObjectType objType, JSType propertyType) {
+      if (!objType.isFunctionType()) {
```

Comment:
Technically, we can get into the same issue with static fields. It is ok to not bother with fields for now, but document this is something that we might have to revisit.